src/api/metadata/index.ts

```diff
@@ -1,24 +1,38 @@
 import log from 'loglevel'
 import { Context } from '../../utils/context'
 import { getSerializedCID, loadIpfsFromCid } from '../../utils/appData'
-import { AppDataDoc } from './types'
+import { pinJSONToIPFS } from '../../utils/ipfs'
+import { AppDataDoc, MetadataDoc } from './types'
 import { CowError } from '../../utils/common'
 
+const DEFAULT_APP_CODE = 'CowSwap'
+
 export class MetadataApi {
   context: Context
 
   constructor(context: Context) {
     this.context = context
   }
 
+  generateAppDataDoc(metadata: MetadataDoc = {}): AppDataDoc {
+    return {
+      version: '0.1.0',
+      appCode: DEFAULT_APP_CODE,
+      metadata: {
+        ...metadata,
+      },
+    }
+  }
+
   async decodeAppData(hash: string): Promise<void | AppDataDoc> {
     try {
       const cidV0 = await getSerializedCID(hash)
       if (!cidV0) throw new CowError('Error getting serialized CID')
-      return await loadIpfsFromCid(cidV0)
-    } catch (error) {
+      return loadIpfsFromCid(cidV0)
+    } catch (e) {
+      const error = e as Error
```

Comment:
Why is it needed to cast as Error?

Reply:
It should be casted as `CowError`. That is due to `e` is typed as `unknown` by default